src/dlgColorTrigger.cpp

```diff
@@ -23,405 +23,430 @@
 #include "dlgColorTrigger.h"
 
 
+#include "dlgTriggerEditor.h"
 #include "Host.h"
 #include "TTextEdit.h"
 #include "TTrigger.h"
 #include "mudlet.h"
 
 
-dlgColorTrigger::dlgColorTrigger(QWidget* pF, TTrigger* pT, int mode) : QDialog(pF), mpTrigger(pT), mMode(mode)
+dlgColorTrigger::dlgColorTrigger(QWidget* pF, TTrigger* pT, const bool isBackGround, const QString& title)
```

Comment:
Old dialog:

![mudlet_353](https://user-images.githubusercontent.com/110988/43043570-b86fa448-8d96-11e8-86fe-877c783fd580.png)

New dialog:

![select foreground trigger color for item 1_352](https://user-images.githubusercontent.com/110988/43043573-bf6f77e6-8d96-11e8-9f66-906af6928c39.png)

The new one is more powerful, but a lot harder to use and less intuitive. We need to design this better.

Reply:
Window title like: Mudlet color selection
Add first line before the three options with the current title as explanation.
Start all 3 options with words like Select or Select from

Reply:
Window title like: Mudlet color selection
Add first line before the three options with the current title as explanation.
Start all 3 options with words like Select or Select from

Reply:
I haven't got ideas on how to improve this yet, but perhaps try reaching out to the community at https://ux.stackexchange.com @SlySven ?

Reply:
This is a short-term *transient* dialogue - and saying what it is doing is consistent with other dialogues in the application - so identifying that it is setting the foreground or background for a particular trigger's item in the title is the right thing, IMHO.

It splits the colour range into the three groups because it is not feasible to provide a push-button for every one of the 256 colours as well as the *default* (the colour used after an `SGR0` and *ignore* (do not pay any attention to this fore/background - only consider the other one) settings.

Ideally I'd like to convert the three `QGroupBox`'s checkboxes into *Radio* buttons so it is more obvious that one and only one is active at a time.  I've partially achieved that by greying out the top set of buttons when that groupbox is deselected and by greying out the text in the other two when they, similarly, deselected (although the latter happens automatically anyhow).

It is useful to report the colour "names" for the first sixteen - because they can be reallocated so that they do not appear as what they are called - but the name is still what the MUD will be thinking is being displayed.

![screenshot_2018-07-22_13-59-23](https://user-images.githubusercontent.com/6163092/43045884-95c2901e-8db8-11e8-82fe-412bcf630cc9.png)

![screenshot_2018-07-22_14-00-02](https://user-images.githubusercontent.com/6163092/43045888-9dc250ba-8db8-11e8-8a33-3d6fb4290549.png)

The current proposal means that:
+ for the basic 16 colour triggers the actions are:
  + (only for an existing one where one of the other 240 colours was used) - click on the group-box checkbox
  + click to choose colour.
+ for a 6x6x6 RxGxB colour:
  + (unless an existing one where one of the other 216 was used) - click on the group-box checkbox
  + click/slide three times to required red/green/blue value
  + click on set button
+ for a grey scale colour:
  + (unless an existing grey scale one where one of the other 23 was used) - click on the group-box checkbox
  + click/slide to required grey value
  + click on set button

Reply:
It is all very logical but it is just way too complicated and I can certainly bet that's what people say of it and give up trying to figure it out. It does need to be simpler to be used without any explanation like the one above was.

Reply:
I have never seen instructions or explanations in window title. Some OS may display differently. Asking again to move this to the inside of the window

Reply:
Oh, apparently macOs applications do not take title bar texts - at least for the `QMessageBox` class:
> `void QMessageBox::setWindowTitle(const QString &title)`
> This function shadows `QWidget::setWindowTitle()`.
> Sets the title of the message box to `title`. On macOS, the window title is ignored (as required by the macOS Guidelines).
>
>This function was introduced in Qt 4.2.

Reply:
Okay, I have added a (smaller text) explanation inside the dialogue I hope that helps...

![screenshot_2018-08-01_19-47-05](https://user-images.githubusercontent.com/6163092/43542035-ce8387ec-95c3-11e8-81fe-2d575c6b0f71.png)